### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by imbcmdth**
Two small fixes here:
* If a PMT has not been seen yet, defer processing PES packets until we have one.
* Clear the PPS and track config between segments so that we handle rendition switching correctly!

### CODE REVIEW

File: lib/transmuxer.js [hunk 1]
```diff
@@ -114,6 +114,8 @@ TransportParseStream = function() {
   TransportParseStream.prototype.init.call(this);
   self = this;
 
+  this.packetsWaitingForPMT = [];
```
Comment (dmlap) [inline lib/transmuxer.js]: I've only been capitalizing the first letter of acronyms when camel-casing them elsewhere (e.g. `startPts`).

File: lib/transmuxer.js [hunk 2]
```diff
@@ -187,8 +189,14 @@ TransportParseStream = function() {
       offset += ((payload[offset + 3] & 0x0F) << 8 | payload[offset + 4]) + 5;
     }
 
+    self.foundPMT = true;
```
Comment (dmlap) [inline lib/transmuxer.js]: Is a separate variable for this necessary? Couldn't we just check `self.programMapTable !== undefined`?

**Comment by dmlap** [comment]
The PPS and track config stuff could use a test.

File: lib/transmuxer.js [hunk 3]
```diff
@@ -114,7 +114,8 @@ TransportParseStream = function() {
   TransportParseStream.prototype.init.call(this);
   self = this;
 
-  this.programMapTable = {};
+  this.packetsWaitingForPMT = [];
```
Comment (dmlap) [inline lib/transmuxer.js]: `s/PMT/Pmt/`

File: lib/transmuxer.js [hunk 4]
```diff
@@ -273,17 +280,26 @@ TransportParseStream = function() {
...
+      this.packetsWaitingForPMT.push([packet, offset, result]);
     } else {
-      result.streamType = this.programMapTable[result.pid];
-      result.type = 'pes';
-      parsePes(packet.subarray(offset), result);
+      this.processPES_(packet, offset, result);
     }
+  };
+
+  this.processPES_ = function (packet, offset, result) {
```
Comment (dmlap) [inline lib/transmuxer.js]: Capitalization.

Comment (imbcmdth) [inline lib/transmuxer.js]: So much nit-picky.

File: lib/transmuxer.js [hunk 5]
```diff
@@ -189,6 +190,11 @@ TransportParseStream = function() {
 
     // record the map on the packet as well
     pmt.programMapTable = self.programMapTable;
+
+    // if there are any packets waiting for a PMT to be found, process them now
+    while (self.packetsWaitingForPMT.length) {
+      self.processPES_.apply(self, self.packetsWaitingForPMT.shift());
```
Comment (dmlap) [inline lib/transmuxer.js]: Why is the `apply` necessary?

File: lib/transmuxer.js (hunk 4 above)
Comment (dmlap) [inline lib/transmuxer.js]: Stop being so sloppy ;)

**Pull request merged**